boa_engine/src/builtins/date/mod.rs

```diff
@@ -1932,3 +1932,16 @@ pub fn this_time_value(value: &JsValue, context: &mut Context) -> JsResult<Date>
         .and_then(|obj| obj.borrow().as_date().copied())
         .ok_or_else(|| context.construct_type_error("'this' is not a Date"))
 }
+
+// TODO perhaps here we can pass in a method that will retrieve the time on a specific platform here
+fn naive_date_time_now() -> NaiveDateTime {
+    chrono::NaiveDateTime::from_timestamp((ic_cdk::api::time() / 1_000_000_000) as i64, 0)
```

Comment:
I can never keep straight which time functions give back which units. Could we add a comment to naive_date_time_now saying that it gives back the current time in seconds or milliseconds or nanoseconds or whatever it is.

Similarly I would vote for changing that 1_000_000_000 out for a variable or a constant called nanoseconds_per_second or whatever it represents.